tst/launch_ci_runner.py

```diff
@@ -0,0 +1,239 @@
+
+        # Update submodules
+        subprocess.run(
+            ["git", "submodule", "update", "--init", "--recursive"], check=True
+        )
+
+        # Run the tests
+        os.chdir(os.path.join(temp_dir, "tst"))
+        build_dir = os.path.join(temp_dir, "build")
+
+        # Run subprocess command to compile code and launch run_tests.py
+        test_command = [
+            "bash",
+            "-c",
+            "source ../env/bash && build_jaybenne -b "
+            + build_dir
+            + " -f && cd "
+            + os.path.join(temp_dir, "tst")
+            + " && ./stepdiff.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff.in"
+            + " --use_mpiexec",
+            +" && ./stepdiff.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_ddmc.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_ddmc.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_ddmc.in --use_mpiexec --mpi_nthreads 8",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_hybrid.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_hybrid.in --use_mpiexec --mpi_nthreads 8",
+        ]
+        ret = subprocess.run(test_command, check=True)
+
+        # Return true if the test script succeeded
+        return ret.returncode == 0
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Run CI tasks with optional Slurm submission."
+    )
+    parser.add_argument(
+        "pr_number", type=int, help="Pull request number for the CI run."
+    )
+    parser.add_argument(
+        "--submission",
+        action="store_true",
+        help="Flag to indicate the script is running as a Slurm submission job.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default=None,
+        help="Output directory created when launching submission script",
+    )
+    args = parser.parse_args()
+
+    # Fetch PR information
+    pr_info = get_pr_info(args.pr_number)
+    head_repo = pr_info["head"]["repo"]["clone_url"]
+    head_ref = pr_info["head"]["ref"]
+    commit_sha = pr_info["head"]["sha"]
+
+    if args.submission:
+        # Update github PR status to indicate we have begun testing
+        update_status(commit_sha, "pending", "CI Slurm job running...")
```

Comment:
Should the the state argument still be  "pending" here, or should it be "submitted"?

Reply:
`"pending"` here is a github-specific string, the only available option for when a job has started running but hasn't completed yet.